### PR TITLE
[WT-129] test: make sure cypress works

### PR DESCRIPTION
**Pull request by JoanVicens**
- Updated sing up test to match new singup page
- Added `data-test` properties to html elements in order to be able to select them on cypress test
- Updated download file service to use `PatchedStreamApi` when is being tested with cypress. Without this the file was never downloaded making the test to always fail.

⚠️ There is a lot of linting going on

### CODE REVIEW

File: src/app/core/components/Navbar/AccountPopover.tsx [hunk 1]
```diff
@@ -64,7 +64,7 @@ export default function AccountPopover({
       </Item>
       <Link
         to="/preferences"
-        className="flex cursor-pointer items-center py-2 px-3 text-gray-80 hover:text-gray-80 hover:bg-gray-1 active:bg-gray-5 no-underline"
+        className="flex cursor-pointer items-center py-2 px-3 text-gray-80 no-underline hover:bg-gray-1 hover:text-gray-80 active:bg-gray-5"
```
Comment (fabioespinosa) [inline src/app/core/components/Navbar/AccountPopover.tsx:67]: Is this from the prettier auto save rule? or is there any change of classes here? 🤔

File: src/app/drive/services/download.service/downloadFile.ts [hunk 2]
```diff
@@ -97,19 +99,22 @@ export default async function downloadFile(
...
+  const isCypress = window['Cypress'] !== undefined;
 
   const writeToFsIsSupported = 'showSaveFilePicker' in window;
   const writableIsSupported = 'WritableStream' in window && streamSaver.WritableStream;
 
   let support: DownloadSupport;
 
-  if (isBrave) {
+  if (isCypress) {
+    support = DownloadSupport.PatchedStreamApi;
```
Comment (fabioespinosa) [inline src/app/drive/services/download.service/downloadFile.ts:116]: Nice find 👌

File: src/app/core/components/Navbar/AccountPopover.tsx (hunk 1 above)
Comment (fabioespinosa) [reply src/app/core/components/Navbar/AccountPopover.tsx:67]: nevermind, it seems its rearranging of existing classes

**Review by fabioespinosa — APPROVED** [review]

File: src/app/drive/services/download.service/downloadFile.ts [hunk 3]
```diff
@@ -57,10 +57,12 @@ function getBlobWritable(filename: string, onClose: (result: Blob) => void): Blo
         closed: Promise.resolve(undefined),
         desiredSize: 3 * 1024 * 1024,
         ready: Promise.resolve(undefined),
-        releaseLock: () => { null; },
+        releaseLock: () => {
+          null;
```
Comment (fabioespinosa) [inline src/app/drive/services/download.service/downloadFile.ts]: Sonar seems to think this is a bug

Comment (fabioespinosa) [reply src/app/drive/services/download.service/downloadFile.ts]: Which makes sense since it is not returning anything

Comment (fabioespinosa) [reply src/app/drive/services/download.service/downloadFile.ts]: Maybe we can remove the null and just comment 

// no op

Comment (redac) [reply src/app/drive/services/download.service/downloadFile.ts]: can we try this?

Comment (fabioespinosa) [reply src/app/drive/services/download.service/downloadFile.ts]: we can! Nice to see a random person commenting on this PR

Comment (fabioespinosa) [reply src/app/drive/services/download.service/downloadFile.ts]: @JoanVicens on the request of @redac :D let's try removing the null and adding // no op

Comment (redac) [reply src/app/drive/services/download.service/downloadFile.ts]: 👋 currently lurking while reading up on the project before contributing.

Comment (fabioespinosa) [reply src/app/drive/services/download.service/downloadFile.ts]: go for it, if you need anything let us know :)

Comment (JoanVicens) [reply src/app/drive/services/download.service/downloadFile.ts]: That made it!

Comment (fabioespinosa) [reply src/app/drive/services/download.service/downloadFile.ts]: Awesome!

Comment (redac) [reply src/app/drive/services/download.service/downloadFile.ts]: > go for it, if you need anything let us know :)

Hi again, it looks like i cannot access the @internxt/inxt-js package even with a valid token in .npmrc when running yarn install.. is this normal?

image

**Pull request merged**